nixos/modules/services/cluster/kubernetes/flannel.nix

```diff
@@ -91,44 +112,69 @@ in
     };
 
     # give flannel som kubernetes rbac permissions if applicable
-    services.kubernetes.addonManager.bootstrapAddons = mkIf ((storageBackend == "kubernetes") && (elem "RBAC" top.apiserver.authorizationMode)) {
-
-      flannel-cr = {
-        apiVersion = "rbac.authorization.k8s.io/v1beta1";
-        kind = "ClusterRole";
-        metadata = { name = "flannel"; };
-        rules = [{
-          apiGroups = [ "" ];
-          resources = [ "pods" ];
-          verbs = [ "get" ];
-        }
-        {
-          apiGroups = [ "" ];
-          resources = [ "nodes" ];
-          verbs = [ "list" "watch" ];
-        }
-        {
-          apiGroups = [ "" ];
-          resources = [ "nodes/status" ];
-          verbs = [ "patch" ];
-        }];
-      };
-
-      flannel-crb = {
-        apiVersion = "rbac.authorization.k8s.io/v1beta1";
-        kind = "ClusterRoleBinding";
-        metadata = { name = "flannel"; };
-        roleRef = {
-          apiGroup = "rbac.authorization.k8s.io";
-          kind = "ClusterRole";
-          name = "flannel";
+    systemd.services.flannel-rbac-bootstrap = mkIf (top.apiserver.enable && (elem "RBAC" top.apiserver.authorizationMode)) {
```

Comment:
First, the storageBackend always was "kubernetes" like declared in line 10 of this file. I created this because in my view the flannel service would need to start before any addons might be deployed to nodes. And nodes couldn't be found before the flannel service configured the network.

Reply:
In https://github.com/NixOS/nixpkgs/pull/56789/commits/6e9037fed0ce0b55ef37188ec1a58e18e196a780 now there is a new service `kube-addon-manager-bootstrap.service` which has the purpose to have higher permissions to invoke the `kubectl apply` and not to conflict with the mkWaitCurl of the `kube-addon-manager.service` which does not need to have clusterAdmin certs but the addonManager certs.

Reply:
I think it is my mistake that I forgot to `mkDefault` the storageBackend in line 40, but that doesn't change the fact that we should *always* be able to handle/support the situation where someone decides to do `storageBackend = mkForce "etcd"`. This is why the mkIf on storageBackend is needed. For the same reason that we cannot assume that `easyCerts` is always true.

Anyway, the new version of the diff looks better. I see why a new service was needed now, because of the order of things starting. :-)